### ISSUE TITLE
Logic to produce nullability fixes should be moved into analyzer_plugin

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/sdk#35872]
See comments at https://dart-review.googlesource.com/c/sdk/+/92151/1/pkg/analysis_server/lib/src/nullability/provisional_api.dart#12

**Comment by bwilkerson** [comment]
> ... it appears that DartEditBuilder only knows how to create new structures, not modify existing structures.

If by "structures" you mean edits, then yes, that's true, and yes, we should talk about the kinds of modifications you think we need and whether it makes sense.

If you mean Dart code, then no. There are methods to replace any range of text by any other text, so it's fully general. We use it pretty much exclusively to support assists, fixes, and have even started using it in refactorings. I'm confident that it's sufficient for these purposes.

**Comment by stereotype441** [comment]
@bwilkerson Were you perhaps talking about `DartFileEditBuilder`?

**Comment by stereotype441** [comment]
@danrubel FYI

**Comment by bwilkerson** [comment]
> Were you perhaps talking about `DartFileEditBuilder`?

Yes. Were you thinking of using `DartEditBuilder` without going through `DartFileEditBuilder`?

**Comment by stereotype441** [comment]
> > Were you perhaps talking about `DartFileEditBuilder`?
> 
> Yes. Were you thinking of using `DartEditBuilder` without going through `DartFileEditBuilder`?

I didn't have any particular plan in mind.  I was just trying to understand what you meant in https://dart-review.googlesource.com/c/sdk/+/92151/1/pkg/analysis_server/lib/src/nullability/provisional_api.dart#12 when you said "add utility methods to `DartEditBuilder`."  Since I wasn't familiar with that class, I didn't know to look for the related class `DartFileEditBuilder`.  Now I understand what you were getting at.

**Comment by bwilkerson** [comment]
I would like to be included in conversations about how to move forward on this as it strongly impacts the work I do.
